### ISSUE TITLE
Document  `iter().copied()` vs `iter()` for small data types

### ISSUE THREAD

**Issue by Dan-wanna-M** [nnethercote/perf-book#83]
As the issues https://github.com/rust-lang/rust/issues/106539 and https://github.com/rust-lang/rust/issues/113789 indicate, iter().copied() is more efficient than iter() for *small* data types because the former one results in less indirections and hence opens up the room for more optimizations. How small is *small* needs more investigations, but anything <= `usize` should definitely be small(in fact, the original reply in the issue suggests `8*usize` is probably still small enough).

**Comment by Shnatsel** [comment]
Indeed, there is a comment form a compiler developer that this is extremely difficult for LLVM to optimize, and that putting `.copied()` early should be encouraged: https://github.com/rust-lang/rust/issues/106539#issuecomment-1374021037

**Comment by nnethercote** [comment]
I added a section about this in [12263da](https://github.com/nnethercote/perf-book/commit/12263da42bfb1058695bf1e9c0844bc238ab7f02). Thanks for the suggestion, and sorry for the slow response!
